### PR TITLE
Add ZKSend wallet implementation and SDK for creating ZKSend links

### PR DESCRIPTION
**Pull request by Jordan-Mysten**
## Description 

Describe the changes or additions included in this PR.

## Test Plan 

How did you test the new or updated feature?

---
If your changes are not user-facing and not a breaking change, you can skip the following section. Otherwise, please indicate what changed, and then add to the Release Notes section as highlighted during the release process.

### Type of Change (Check all that apply)

- [ ] protocol change
- [ ] user-visible impact
- [ ] breaking change for a client SDKs
- [ ] breaking change for FNs (FN binary must upgrade)
- [ ] breaking change for validators or node operators (must upgrade binaries)
- [ ] breaking change for on-chain data layout
- [ ] necessitate either a data wipe or data migration

### CODE REVIEW

File: sdk/zksend/package.json [hunk 1]
```diff
@@ -41,11 +41,18 @@
 	"homepage": "https://github.com/mystenlabs/sui#readme",
 	"devDependencies": {
 		"@mysten/build-scripts": "workspace:*",
+		"@mysten/sui.js": "workspace:*",
 		"@types/node": "^20.4.2",
 		"typescript": "^5.1.6",
 		"vitest": "^0.33.0"
 	},
+	"peerDependencies": {
```
Comment (Jordan-Mysten) [inline sdk/zksend/package.json:49]: Last time I checked the peer deps acted weird on pre-1.0 versions

File: sdk/zksend/src/channel/index.ts [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+			popup?.close();
+			window.removeEventListener('message', listener);
+		};
+
+		window.addEventListener('message', listener);
+
+		popup = window.open(
+			`${origin}/dapp/${type}?${new URLSearchParams({
+				id: this.#id,
+				origin: this.#origin,
```
Comment (Jordan-Mysten) [inline sdk/zksend/src/channel/index.ts:56]: Can you also add a `name` or `title` prop that is supposed to be the name of the dapp? I realized I missed this but it's in the designs.

File: sdk/zksend/src/channel/index.ts [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+
+		return promise;
+	}
+
+	close() {
+		this.#close?.();
+	}
+}
+
+export class ZkSendHost {
```
Comment (Jordan-Mysten) [inline sdk/zksend/src/channel/index.ts:72]: We'll also need methods on this to parse the request

File: sdk/zksend/src/wallet.ts [hunk 4]
```diff
@@ -0,0 +1,186 @@
...
+		this.#accounts = [];
+		this.#events = mitt();
+		this.#client = suiClient;
+	}
+
+	#signTransactionBlock: SuiSignTransactionBlockMethod = async ({ transactionBlock, account }) => {
+		transactionBlock.setSenderIfNotSet(account.address);
+
+		const bytes = toB64(
+			await transactionBlock.build({
```
Comment (Jordan-Mysten) [inline sdk/zksend/src/wallet.ts:97]: Shouldn't this use `serialize` instead and have the zkSend dapp build to bytes? That's how other wallets work, at least. Not strongly tied to this tho.

File: sdk/zksend/src/wallet.ts [hunk 5]
```diff
@@ -0,0 +1,186 @@
...
+		return () => this.#events.off(event, listener);
+	};
+
+	#setAccount(address?: string) {
+		if (address) {
+			this.#accounts = [
+				new ReadonlyWalletAccount({
+					address,
+					chains: [SUI_MAINNET_CHAIN],
+					features: ['sui:signTransactionBlock'],
```
Comment (Jordan-Mysten) [inline sdk/zksend/src/wallet.ts:139]: I think this should also have sign message?

**Review by hayes-mysten — APPROVED** [review]

**Pull request merged**